Add document to showcase scaleout on hpu (#14357)
Lightning-AI/lightning: docs/source-pytorch/accelerators/hpu_basic.rst

----

> Scale-out on Gaudis

Reviewer: ```suggestion
Run across many machines
```
Or something similar? The term "scale-out" is lesser known imo.

Author: I've used it for distributed training in hpu docs, so aligning with the same.

-------------------

> To train a Lightning model using multiple HPU nodes, set the ``num_nodes`` parameter with the available nodes in the Trainer class

Reviewer 2: ```suggestion
To train a Lightning model using multiple HPU nodes, set the ``num_nodes`` parameter with the available nodes in the ``Trainer`` class.
```

.. code-block:: python

    trainer = Trainer(accelerator="hpu", devices=8, strategy="hpu_parallel", num_nodes=2)

> In addition to this, the following environment variables needs to be set to establish communication across nodes. Check out the documentation on ` Cluster <https://pytorch-lightning.readthedocs.io/en/stable/clouds/cluster.html>`__ for more details.

Reviewer 2: ```suggestion
In addition to this, the following environment variables need to be set to establish communication across nodes. Check out the documentation on :doc:`Cluster Environment <../clouds/cluster>` for more details.
```

- *MASTER_PORT* - required; has to be a free port on machine with NODE_RANK 0
- *MASTER_ADDR* - required (except for NODE_RANK 0); address of NODE_RANK 0 node
- *WORLD_SIZE* - required; how many workers are in the cluster
- *NODE_RANK* - required; id of the node in the cluster

> The trainer needs to be instantiated on every node that is participating in the training

Reviewer 2: ```suggestion
The trainer needs to be instantiated on every node participating in the training.
```
